tests/test_xarray_backend.py

```diff
@@ -0,0 +1,405 @@
+"""Test titiler-cmr xarray backend."""
+
+import io
+from copy import deepcopy
+from datetime import datetime, timedelta
+from math import ceil
+from pathlib import Path
+from typing import Tuple
+from urllib.parse import urlencode
+
+import pytest
+from fastapi.testclient import TestClient
+from geojson_pydantic import Feature, Polygon
+from httpx import Response
+from PIL import Image
+
+from titiler.cmr.timeseries import TimeseriesMediaType
+from titiler.core.models.mapbox import TileJSON
+
+
+@pytest.mark.vcr
+def test_xarray_tilejson(app, xarray_query_params):
+    """Test /tilejson.json endpoint for xarray backend"""
+
+    response = app.get(
+        "/WebMercatorQuad/tilejson.json",
+        params={
+            **xarray_query_params(),
+            "datetime": "2024-10-11T00:00:00Z/2024-10-12T23:59:59Z",
+        },
+    )
+
+    assert response.status_code == 200
+    assert response.headers["content-type"] == "application/json"
+
+    tilejson = response.json()
+    assert tilejson["bounds"] == [-180.0, -90.0, 180.0, 90.0]
+
+
+@pytest.mark.vcr
+def test_xarray_tilejson_with_sel(app, xarray_query_params):
```

Comment:
I added this test, otherwise tests are unchanged apart from moving them to separate files (test_rasterio_backend and test_xarray_backend)